harness/tests/test_horovod.py

```diff
@@ -70,7 +70,7 @@ def test_create_run_command(
 
 def test_create_hostlist_arg() -> None:
     ip_addresses = ["localhost", "128.140.2.4"]
-    num_proc_per_machine = 8
+    num_proc_per_machine = [8, 8]
     expected_horovod_hostlist_arg = (
         f"{ip_addresses[0]}:{num_proc_per_machine},{ip_addresses[1]}:{num_proc_per_machine}"
     )
```

Comment:
this looks like it couldn't possibly pass.

Also I'd recommend just hardcoding the expect string as "localhost:8,128.140.2.4:8" which would be easier to read anyway.